pkg/asset/installconfig/aws/validation.go

```diff
@@ -129,6 +129,10 @@ func validateMachinePool(ctx context.Context, meta *Metadata, fldPath *field.Pat
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("zones"), pool.Zones, errMsg))
 		}
 	}
+	if pool.InstanceType != "" {
+		session, _ := meta.Session(ctx)
```

Comment:
So currently this validation package uses [`meta *Metadata`](https://github.com/openshift/installer/pull/4258/files#diff-784272713154e11b21c479afd5c0ad47R107) to get information from the AWS API..

see https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/metadata.go#L19-L20 being used in validation here https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/validation.go#L75

We can add a private function to load the instance types on the metadata maybe all for the time being, and then we can use those to also unit tests the validation code.

I think you can use the public / private subnet handling for example.

Reply:
I had considered using metadata but I wasn't sure if extending the metadata was a good idea.  I'll rework this to use a function to query all instance types then do the validation.

Reply:
Updated.